### PR TITLE
CODENVY-1112: Add workspces number limit check, when starting workspace from config

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Add workspces number limit check, when starting workspace from config.

### What issues does this PR fix or reference?
https://github.com/codenvy/codenvy/issues/1112

### Previous Behavior
It was allowed to create infinite number of workspaces by starting workspaces from config (from swagger or other rest client).

### New Behavior
Creating workspace when starting workspace from config is controlled by `limits.user.workspaces.count` property.

### Tests written?
Yes

@evoevodin please review

### CODE REVIEW

File: wsmaster/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 1]
```diff
@@ -236,12 +237,17 @@ public WorkspaceImpl updateWorkspace(String id, Workspace update) throws Conflic
...
      */
     @VisibleForTesting
-    <T extends WorkspaceImpl> T checkCountAndPropagateCreation(String namespace,
-                                                               WorkspaceCallback<T> callback) throws ServerException,
+    <T extends WorkspaceImpl> T checkNumberOfWorkspacesAndPropagateCreation(String namespace,
+                                                                            WorkspaceCallback<T> callback) throws ServerException,
                                                                                                      NotFoundException,
                                                                                                      ConflictException {
+        checkNumberOfWorkspaces(namespace);
+        return callback.call();
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: -1 to merge it. Looks like you create workspace out of lock

**Comment by vinokurig** [comment]
@sleshchenko I have reworked it

File: wsmaster/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 2]
```diff
@@ -270,10 +271,7 @@ void checkMaxEnvironmentRam(WorkspaceConfig config) throws ServerException {
             Environment env = envEntry.getValue();
             final long workspaceRam = sumRam(env);
             if (workspaceRam > maxRamPerEnvMB) {
-                throw new LimitExceededException(format("The maximum RAM per workspace is set to '%dmb' and you requested '%dmb'. " +
-                                                        "This value is set by your admin with the 'limits.workspace.env.ram' property",
-                                                        maxRamPerEnvMB,
-                                                        workspaceRam),
+                throw new LimitExceededException(format("You are only allowed to use %d mb. RAM per workspace.", maxRamPerEnvMB),
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: Why did you change this message?

**Comment by sleshchenko** [comment]
Also please fix commit message

File: wsmaster/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java (hunk 2 above)
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: OK. Thanks for explanation
